style-packages.md

```diff
@@ -0,0 +1,255 @@
+
+``` wrong
+github.com/user/repo/src/httputil   // DON'T DO IT, AVOID SRC!!
+
+github.com/user/repo/gosrc/httputil // DON'T DO IT, AVOID GOSRC!!
+```
+
+### No plurals
+
+In go, package names are not plural. This is surprising to programmers who came
+from other languages and are retaining an old habit of pluralizing names.
+Don't name a package httputils, but httputil!
+
+``` wrong
+package httputils  // DON'T DO IT, USE SINGULAR FORM!!
+```
+
+### Renames should follow the same rules
+
+If you are importing more than one packages with the same name, you can locally
+rename the package names. The renames should follow the same rules mentioned
+on this article. There is no rule which package you should rename. If you are
+renaming the standard package library, it is nice to add a go prefix to make the name
+self document that it is "Go standard library's" package, e.g. `gourl`, `goioutil`.
+
+```
+import (
+    gourl "net/url"
+
+    "myother.com/url"
+)
+```
+
+### Enforce vanity URLs
+
+`go get` supports getting packages by a URL that is different than the URL
+of the package's repo. These URLs are called vanity URLs and require you to
+serve a page with specific meta tags the Go tools recognize.
+You can serve a package with a custom domain and path using vanity URLs.
+
+For example,
+
+```
+$ go get cloud.google.com/go/datastore
+```
+
+checks out the source code from `https://code.googlesource.com/gocloud` behind
+the scenes and puts it in your workspace under $GOPATH/src/cloud.google.com/go/datastore.
+
+Given code.googlesource.com/gocloud is already serving this package, would it
+be possible to go get the package from that URL? The answer is no, if you enforce
+the vanity URL.
+
+To do that, add an import statement to the package. The go tool will reject
+any import of this package from any other path and will display a friendly
+error to the user. If you don't enforce your vanity URLs, there will be two
+copies of your package that cannot work together due to the different namespace.
+
+```
+package datastore // import "cloud.google.com/go/datastore"
+```
+
+----
+
+## Package-level documentation
+
+Always document the package. Package documentation is a top-level comment
+immediately preceding the package clause. For non-main packages, godoc always starts with
+"Package {pkgname}" and follows with a description. For main packages, godoc
+can start with "Command {pkgname}", "Program {pkgname}", etc.
+
+```
+// Package ioutil implements some I/O utility functions.
+package ioutil
+
+// Command gops lists all the processes running on your system.
+package main
+
+// Sample helloworld demonstrates how to use x.
+package main
+```
+
+### Use doc.go
+
+Sometimes, package docs can get very lengthy, especially when they provide details
+of usage and guidelines.
+Move the package godoc to a `doc.go` file.
+(See an example of a [doc.go](https://github.com/GoogleCloudPlatform/google-cloud-go/blob/master/datastore/doc.go).)
+
+----
+
```

Comment:
I couldn't come up with a small summary that will be essential and enough. I think package organization requires a guideline on its own. I don't want to mention a few ideas here and leaving the importing points out. Let's keep the scope to a single package on this doc.